packages/odyssey-react-mui/src/theme/useContrastMode.test.tsx

```diff
@@ -0,0 +1,254 @@
+import { render, waitFor, act } from "@testing-library/react";
+import {
+  ContrastModeContext,
+  useContrastMode,
+  useContrastModeContext,
+  getBackgroundColor,
+  hueNeutral50Rgb,
+  type ContrastModeContextType,
+} from "../useContrastMode";
+import * as Tokens from "@okta/odyssey-design-tokens";
+import { renderHook } from "@testing-library/react";
+
+// Common wrapper component for tests
+const createWrapper = (contextValue: ContrastModeContextType) => {
+  return ({ children }: { children: React.ReactNode }) => (
+    <ContrastModeContext.Provider value={contextValue}>
+      {children}
+    </ContrastModeContext.Provider>
+  );
+};
+
+describe("useContrastMode and related functions", () => {
+  const originalGetComputedStyle = window.getComputedStyle;
+
+  beforeEach(() => {
+    window.getComputedStyle = jest.fn().mockImplementation(() => ({
+      backgroundColor: "rgba(0, 0, 0, 0)",
+    }));
+  });
+
+  afterEach(() => {
+    window.getComputedStyle = originalGetComputedStyle;
+    jest.clearAllMocks();
+    document.documentElement.style.backgroundColor = "";
+  });
+
+  describe("useContrastMode hook", () => {
+    it("should return lowContrast mode by default", () => {
+      const { result } = renderHook(() => useContrastMode({}), {
+        wrapper: createWrapper({ contrastMode: "lowContrast" }),
+      });
+
+      expect(result.current.contrastMode).toBe("lowContrast");
+      expect(result.current.contrastContainerRef.current).toBe(null);
+      expect(result.current.parentBackgroundColor).toBe("#ffffff");
+    });
+
+    it("should respect explicitly set contrast mode", () => {
+      const { result } = renderHook(
+        () => useContrastMode({ contrastMode: "highContrast" }),
+        {
+          wrapper: createWrapper({ contrastMode: "lowContrast" }),
+        },
+      );
+
+      expect(result.current.contrastMode).toBe("highContrast");
+    });
+
+    it("should update contrast mode based on background color changes", async () => {
+      const TestComponent = () => {
+        const { contrastContainerRef, contrastMode } = useContrastMode({});
+        return (
+          <div ref={contrastContainerRef} data-testid="container">
+            {contrastMode}
+          </div>
+        );
+      };
+
+      // Mock getComputedStyle before rendering
+      window.getComputedStyle = jest.fn().mockImplementation(() => ({
+        backgroundColor: Tokens.HueNeutral50,
+      }));
+
+      const { getByTestId } = render(
+        <ContrastModeContext.Provider value={{ contrastMode: "lowContrast" }}>
+          <TestComponent />
+        </ContrastModeContext.Provider>,
+      );
+
+      const testContainer = getByTestId("container");
+
+      act(() => {
+        testContainer.style.backgroundColor = Tokens.HueNeutral50;
+        const event = new Event("transitionend");
+        Object.defineProperty(event, "propertyName", {
+          value: "background-color",
+        });
+        testContainer.dispatchEvent(event);
```

Comment:
I think there's a better way to write this without using `Object.defineProperty`.

Reply:
We should also comment this to explain why we're going through the trouble of firing the `transitionend` event.

It might not be an issue with Happy-DOM; that, or we need to wrap it in a `setTimeout` in a `new Promise` and make this test async possibly.

Reply:
Maybe we can simply make the `act` async.